# ðŸŒ¼ðŸŒ¼ðŸŒ¼ mitogenomeç»„è£…ç®€å•ç‰ˆ ðŸŒ¼ðŸŒ¼ðŸŒ¼
source activate getorganelle
cd /data01/xush/1.mitogenome/0.data
for i in *_1.fq.gz
do
i=${i%_1.fq.gz};
extractfq -fq1 ${i}_1.fq.gz \
          -fq2 ${i}_2.fq.gz \
	  -outfq1 ${i}_1_extractfq.fq.gz \ 
          -outfq2 ${i}_2_extractfq.fq.gz \
	  -size_required 2 -gz
done
---------------------------------------------------
å®‰è£…mitozçš„çŽ¯å¢ƒï¼šå…ˆä¸‹è½½åŽ‹ç¼©åŒ…ï¼Œç„¶åŽæ¿€æ´»çŽ¯å¢ƒ
mkdir -p ./mitoz3.6
tar -xzf ./mitoz3.6.tar.gz -C ./mitoz3.6
ä¸‹è½½åœ°å€ï¼šhttps://www.dropbox.com/scl/fo/4md8irodd9flywxhp85wf/AEXYFykNDPgVuJ6nSS6CCjs?rlkey=gpouy6vue1rf9dgva4jqqcik0&e=1&dl=0 #ç›´æŽ¥ä¸‹è½½
source /data01/xush/1.mitogenome/mitoz3.6/bin/activate
conda-unpack
---------------------------------------------------
source /data01/xush/1.mitogenome/mitoz3.6/bin/activate
cd /data01/xush/1.mitogenome/0.data
for i in ../0.data/*_1_extractfq.fq.gz; do
    filename=$(basename "$i")              # å–å‡ºæ–‡ä»¶åéƒ¨åˆ†
    sample=${filename%_1_extractfq.fq.gz}  # åŽ»æŽ‰åŽç¼€å˜æˆæ ·æœ¬å
    mitoz all \
    --outprefix ${sample} \
    --thread_number 20 \
    --clade Chordata \
    --genetic_code 2 \
    --species_name "Sinocyclocheilus" \
    --fq1 ../0.data/${sample}_1_extractfq.fq.gz \
    --fq2 ../0.data/${sample}_2_extractfq.fq.gz \
    --fastq_read_length 150 \
    --assembler megahit \
    --requiring_taxa Cyprinidae \
    --workdir /data01/xush/1.mitogenome/2.results/${sample}_workdir
done
#æˆ–è€…ä¸é€šè¿‡ä¸Šé¢çš„ç›´æŽ¥åŠ å…¥å‚æ•°--data_size_for_mt_assembly 3 
#âœ…å¦‚æžœæ˜¯è„Šæ¤ŽåŠ¨ç‰©ï¼ˆå¦‚é±¼ç±»ã€å“ºä¹³ç±»ã€é¸Ÿç±»ç­‰ï¼‰ï¼š
--genetic_code 2
ä»£è¡¨ Vertebrate Mitochondrial Codeï¼ˆè„Šæ¤ŽåŠ¨ç‰©çº¿ç²’ä½“é—ä¼ å¯†ç ï¼‰
âœ…å¦‚æžœæ˜¯æ— è„Šæ¤ŽåŠ¨ç‰©ï¼š
æ˜†è™«ï¼š--genetic_code 5 ï¼ˆInvertebrate Mitochondrialï¼‰
å…¶ä»–å…·ä½“åˆ†ç±»å¯èƒ½ä¸åŒï¼Œå¯ä»¥æŸ¥ï¼šNCBI genetic code tables
---------------------------------------------


---------------------------------------------
# ðŸŒ¼ðŸŒ¼ðŸŒ¼mitogenomeç»„è£…å¤æ‚ç‰ˆðŸŒ¼ðŸŒ¼ðŸŒ¼
# è´¨æŽ§ï¼Œfastqc *.fq.gzï¼Œcleandataçš„è´¨é‡è¾ƒå¥½
# ç”¨bwaè¿›è¡ŒMapping åŸºå› ç»„é€‰æ‹©è¿‘ç¼˜ç‰©ç§çš„åŸºå› ç»„
cd /data01/xush/1.mitogenome/1.mapping/ref
bwa index Sang-hap1.chr.fasta

cd /data01/xush/1.mitogenome/0.data
input=/data01/xush/1.mitogenome/0.data
outdir=/data01/xush/1.mitogenome/1.mapping
ref=/data01/xush/1.mitogenome/1.mapping/ref/Sang-hap1.chr.fasta

for i in *_1.fq.gz; do
    sample=${i%_1.fq.gz}
    fq1=${sample}_1.fq.gz
    fq2=${sample}_2.fq.gz
    bwa mem -t 10 -M -R "@RG\tID:$sample\tPL:illumina\tLB:$sample\tSM:$sample" \
        $ref $input/$fq1 $input/$fq2 | \
    samtools view -Sb -o $outdir/${sample}.bam
done
----------------------------------------------------------------------
## çº¿ç²’ä½“ç»„è£…æ³¨é‡Šæµç¨‹ ##
----------------------------------------------------------------------
# Sang-hap1.chr.fastaä¸­æ²¡æœ‰çº¿ç²’ä½“åºåˆ—
# mappingå¾—åˆ°bamæ–‡ä»¶ï¼ŒæŠŠå…¶ä¸­unmapçš„éƒ¨åˆ†æå–å‡ºæ¥ï¼šï¼ˆ4+8=12ï¼Œä¸¤ç«¯éƒ½æ²¡æ¯”å¯¹ä¸Šçš„ï¼‰
# ç„¶åŽæŒ‰åç§°æŽ’åºï¼Œç„¶åŽç”¨bedtoolså°†bamè½¬ä¸ºfastq
	#!/bin/bash
	cd /work/users/wangyw/02_S.micro/output/bam
	for i in `ls *bam`;
	do
	samtools view -bf 12 $i > ${i%%.*}-unmap.bam
	samtools sort -n ${i%%.*}-unmap.bam -o ${i%%.*}-unmap-namesort.bam
	bedtools bamtofastq -i ${i%%.*}-unmap2-namesort.bam -fq ../mt_fq/${i%%.*}-unmap_r1.fq -fq2 ../mt_fq/${i%%.*}-unmap_r2.fq
	rm ${i%%.*}-unmap.bam
	rm ${i%%.*}-unmap-namesort.bam
	done
----------------------------------------------------------------------
#### æŠŠunmapçš„fqåºåˆ—å†mapåˆ°çº¿ç²’ä½“åŸºå› ç»„ä¸Š(ä»Žç½‘ä¸Šä¸‹è½½å°çœ¼é‡‘çº¿é²ƒçº¿ç²’ä½“åŸºå› ç»„ GenBank: MN145877.1)
	#!/bin/bash
	mydir=/work/users/wangyw/02_S.micro
	genome=$mydir/input/mt/S.micro.fa
	index=$mydir/input/mt/S.micro.fa

	cd $mydir/output/mt_fq

	for i in `ls *unmap_r1.fq`
	do
	fq1=$i
	fq2=${i%-*}-unmap_r2.fq
	sample=${i%-*}

	bwa mem -M -R "@RG\tID:$sample\tPL:illumina\tLB:$sample\tSM:$sample" $index $fq1 $fq2 | samtools view -Sbh -o $mydir/output/mt_bam/$sample.bam
	samtools flagstat $mydir/output/mt_bam/$sample.bam > $mydir/output/mt_bam/${sample}.flagstat
	done
#ç»Ÿè®¡
for i  in `ls *.flagstat`;do reads=`awk 'NR==5' $i` ;echo $i $reads >>mt-raw-stat;done
#åˆ é™¤unmap.fq
----------------------------------------------------------------------
#### æå–çº¿ç²’ä½“åºåˆ—ï¼š
	#å…ˆè¿‡æ»¤:
	cd /work/users/wangyw/02_S.micro/output/mt_bam
	for i in `ls *.bam`
	do 
	samtools view -b -f 2 -F 256 -q 30 $i -o ${i%%.*}-f2F256q30.bam
	samtools flagstat ${i%%.*}-f2F256q30.bam > ../mt_bamflagstat/${i%%.*}-f2F256q30.flagstat
	done
	#ç»Ÿè®¡
	for i  in `ls *-f2F256q30.flagstat`;do reads=`awk 'NR==5' $i` ;echo $i $reads >>mt-filt-stat;done
	# åˆ é™¤*.bam

	#æå–çº¿ç²’ä½“åºåˆ—ï¼Œæ–¹æ³•åŒä¸Šã€‚
	cd /work/users/wangyw/02_S.micro/output/mt_bam
	for i in `ls *f2F256q30.bam`;
	do
	samtools sort -n $i -o ${i%%.*}-namesort.bam
	bedtools bamtofastq -i ${i%%.*}-namesort.bam -fq ../mt_fq/${i%%.*}-mt_r1.fq -fq2 ../mt_fq/${i%%.*}-mt_r2.fq
	rm ${i%%.*}-namesort.bam
	done
----------------------------------------------------------------------
#### ç»„è£…çº¿ç²’ä½“åŸºå› ç»„ï¼š
#ä½¿ç”¨MitoZæ¥ç»„è£…ï¼ˆhttps://github.com/linzhi2013/MitoZï¼‰
	#ç»„è£…
	(mitozEnv) [wangyw@Buffalo mt_mitoz]$ cat mt_mitoz_all.sh
	mydir=/work/users/wangyw/02_S.micro
	fq1=$1
	fq2=$2
	outfile=$3

	python3 /work/users/wangyw/biosoft/release_MitoZ_v2.3/MitoZ.py all2 \
	--genetic_code 2 \
	--clade Chordata \
	--insert_size 350 \
	--thread_number 8 \
	--fastq1 $fq1 \
	--fastq2 $fq2 \
	--outprefix $outfile \
	--fastq_read_length 150 \
	--run_mode 2 \
	1>${outfile}.log 2>${outfile}.err

	if [[ ! -d "$tmp" ]];then mv tmp ${outfile}.tmp ;fi

	#æ‰¹é‡ç»„è£…è„šæœ¬
	(mitozEnv) [wangyw@Buffalo mt_mitoz]$ cat all.sh
	for i in `ls /work/users/wangyw/02_S.micro/output/mt_fq/*-f2F256q30-mt_r1.fq`
	do
	filen1=${i%-*};
	filen2=${filen1##*/}
	filen3=${filen2%-*}
	sh mt_mitoz_all.sh ${i%_*}_r1.fq ${i%_*}_r2.fq $filen3
	done

#å…ˆè¿›å…¥çŽ¯å¢ƒï¼Œç„¶åŽå¼€å§‹ç»„è£…
source activate mitozEnv
nohup sh all.sh > all.sh.nohup 2>&1 &


##SCP-10æ²¡æœ‰ç»“æžœï¼Œå› ä¸ºæ•°æ®è¾ƒå°‘? 
#æ¢ç”¨å°çš„kmeræ¥ç»„è£…
source activate mitozEnv
(mitozEnv) [wangyw@Buffalo SCP-10.assembly]$ pwd
/work/users/wangyw/02_S.micro/output/tmp/SCP-10.assembly
(mitozEnv) [wangyw@Buffalo SCP-10.assembly]$ /work/users/wangyw/biosoft/release_MitoZ_v2.3/bin/assemble/mitoAssemble all -K 51 -o work51 -s work71.soaptrans.lib -p 8
#æå–å‡ºmito
(mitozEnv) [wangyw@Buffalo SCP-10.assembly]$ cat findmitoscaf.sh
mydir=/work/users/wangyw/02_S.micro
fq1=$mydir/output/mt_fq/SCP-10-f2F256q30-mt_r1.fq
fq2=$mydir/output/mt_fq/SCP-10-f2F256q30-mt_r2.fq
python3 /work/users/wangyw/biosoft/release_MitoZ_v2.3/MitoZ.py findmitoscaf --genetic_code 2 --clade Chordata \
--outprefix test \
--thread_number 8 \
--fastq1 $fq1 \
--fastq2 $fq2 \
--fastq_read_length 150 \
--fastafile work51.scafSeq
#æ³¨é‡Š
python3 /work/users/wangyw/biosoft/release_MitoZ_v2.3/MitoZ.py annotate --genetic_code 2 --clade Chordata \
--outprefix SCP-10-test --thread_number 8 \
--fastafile test.mitogenome.fa
#æœ€åŽçš„ç»“æžœæ–‡ä»¶ä¸ºSCP-10-test.fasta
----------------------------------------------------------------------

#ä½¿ç”¨MitoSæ¥æ³¨é‡Š
#ç½‘å€ï¼šhttp://mitos2.bioinf.uni-leipzig.de/index.py

#ä¸€äº›éœ€è¦è¿›è¡Œåå‘äº’è¡¥
source activate mitozEnv
for i in `ls *fasta`;do perl reverse_complementary.pl -fa $i -out ${i%.*}-reverse.fasta; done

#ç„¶åŽé‡æŽ’ï¼ˆæŒ‰ç…§å‚è€ƒåŸºå› ç»„ï¼‰
for i in `ls *fasta`
do
python3 /work/users/wangyw/biosoft/release_MitoZ_v2.3/useful_scripts/Mitogenome_reorder.py \
-f $i \
-r /work/users/wangyw/02_S.micro/input/mt/S.micro.fa
done

----------------------------------------------------------------------
#è¿˜æœ‰ä¸€äº›éœ€è¦æ¢ç”¨å°çš„kmeræ¥ç»„è£…
	#æ²¡æœ‰æˆçŽ¯ï¼š
	JYP-03 --kmer51
	SCP-06 --kmer51
	SMHP-02 --kmer51
	ZZP-02.fasta --kmer51
	#æ³¨é‡Šæœ‰é—®é¢˜ï¼š
	FYP-03ï¼šnotfound-trnT --kmer51
	JYP-15ï¼šSplit/duplicated features: cob --kmer31

----------------------------------------------------------------------
#è§£åŽ‹
for i in `ls`;do mkdir ../anno/${i%.*};done
for i in `ls`;do unzip $i -d ../anno/${i%.*};done
#æŠŠé»˜è®¤çš„æ–‡ä»¶åæ”¹ä¸ºæ ·æœ¬å
for i in `ls`;do cd $i; cd `ls`;cp result.fas ../../../anno-fas/$i.fas;cd ../..;done

## ä¸ºåºåˆ—åå¡«åŠ ä¸Šæ ·æœ¬å
for i in `ls`;do cat $i|sed "s/>/>${i%.*};/g" > ../anno-fas2/$i ;done

----------------------------------------------------------------------

